plugins/qtAccessionPlugin/modules/accession/actions/eventsComponent.class.php

```diff
@@ -0,0 +1,147 @@
+    foreach (QubitTerm::get($criteria) as $term)
+    {
+      $this->eventTypes[$term->id] = $term->getName(['cultureFallback' => true]);
+    }
+
+    // Define form used to add/edit events
+    $this->form = new sfForm;
+    $this->form->getValidatorSchema()->setOption('allow_extra_fields', true);
+
+    $this->addField('eventType');
+    $this->addField('date');
+    $this->addField('agent');
+    $this->addField('note');
+
+    // Summarize/cache existing alternative identifier data
+    $this->eventData = [];
+
+    foreach ($this->resource->getEvents() as $event)
+    {
+      $property = $event->getPropertyByName('accessionEventNote');
+
+      $this->eventData[] = [
+        'id' => $event->id,
+        'typeId' => $event->typeId,
+        'date' => $event->getDate(['sourceCulture' => true]),
+        'agent' => $event->getAgent(),
+        'object' => $event,
+        'note' => $property,
+        'accessionId' => $this->resource->id
+      ];
+    }
+  }
+
+  protected function addField($name)
+  {
+    switch ($name)
+    {
+    case 'eventType':
+        $this->form->setValidator($name, new sfValidatorInteger);
+        $this->form->setWidget($name, new sfWidgetFormSelect(['choices' => $this->eventTypes]));
+
+        break;
+
+      case 'date':
+        $this->form->setValidator($name, new sfValidatorString);
+        $widget = new sfWidgetFormInput(['label' => false]);  
+        $this->form->setWidget($name, $widget);
+
+        break;
+
+      case 'agent':
+        $this->form->setValidator($name, new sfValidatorString);
+        $widget = new sfWidgetFormInput(['label' => false]);
+        $this->form->setWidget($name, $widget);
+
+        break;
+
+      case 'note':
+        $this->form->setValidator($name, new sfValidatorString);
+        $widget = new sfWidgetFormTextarea(['label' => false]);
+        $widget->setAttribute('placeholder', $this->context->i18n->__('Notes'));
+        $this->form->setWidget($name, $widget);
+
+        break;
+    }
+  }
+
+  public function processForm()
+  {
+    $finalEvents = [];
+
+    if (is_array($this->request->events))
+    {
+      foreach ($this->request->events as $item)
+      {
+        // Continue only if event type is populated
+        if (empty($item['eventType']) || empty($item['date']))
+        {
+          continue;
+        }
+
+        // Fetch or create new accession event object
+        if (!empty($item['id']))
+        {
+          $finalEvents[] = $item['id'];
+
+          $event = QubitAccessionEvent::getById($item['id']);
+        }
+        else
+        {
+          $event = new QubitAccessionEvent;
+        }
+
+        $event->accessionId = $this->resource->id;
+        $event->typeId = $item['eventType'];
+        $event->date = empty($item['date']) ? null : $item['date'];
+        $event->agent = $item['agent'];
+        $event->save();
+
+        // Store note as property
+        $property = $event->getPropertyByName('accessionEventNote');
```

Comment:
Why not use the `note` table here instead of the `property` table?

Reply:
I think the CAAIS spec allows for just one note (http://archivescanada.ca/CWG_AccessionStandard#5.1_e) so I was using property to keep it simple, but I could do it with note too. I guess using note would be more future proof in case they update the spec.